pyflakes/checker.py

```diff
@@ -1575,7 +1607,27 @@ def BINOP(self, node):
             self._handle_percent_format(node)
         self.handleChildren(node)
 
-    NUM = STR = BYTES = ELLIPSIS = CONSTANT = ignore
+    def STR(self, node):
+        if self._in_annotation and not self._in_typing_literal:
+            fn = functools.partial(
+                self.handleStringAnnotation,
+                node.s,
+                node,
+                node.lineno,
+                node.col_offset,
+                messages.ForwardAnnotationSyntaxError,
+            )
+            if self._in_deferred:
+                fn()
+            else:
+                self.deferFunction(fn)
+
+    if PY38_PLUS:
+        def CONSTANT(self, node):
+            if isinstance(node.value, str):
+                return self.STR(node)
+    else:
+        NUM = BYTES = ELLIPSIS = CONSTANT = ignore
```

Comment:
Just checking it's intentional that `NUM`, `BYTES` and `ELLIPSIS` aren't defined at all in the `PY38_PLUS` branch here? (I'm not sure I follow exactly what these methods do, just highlighting the imbalance) I'm guessing in later versions of Python they're all folded into `CONSTANT`?

Reply:
yep, that was one of the major changes of python3.8's ast -- `ast.Constant` now represents all of the different types of constants